### PR TITLE
Sowmitranalla/pp 226 upgrade to hapi v17

### PR DESCRIPTION
**Pull request by sowmitranalla**
- this PR adds compatibility for Hapi v17
- All the build tooling has decayed and needed updating
- Bumps Node.js to LTS
- Bumps all dependencies
- This is a SemVer MAJOR update
- migrated from yarn to npm 
- removed coveralls

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,14 +28,14 @@
...
-    "eslint-config-lob": "^3.0.0",
-    "hapi": "^16.0.0",
-    "jshint-stylish": "^1.0.0",
-    "mocha": "^5.2.0",
-    "nyc": "^13.1.0"
+    "chai": "^4.3.4",
+    "eslint": "^7.32.0",
+    "eslint-config-lob": "^5.2.0",
+    "hapi": "^17.8.5",
+    "joi": "^10.6.0",
```
Comment (Lianathanoj) [inline package.json:35]: see if you can bump joi up to `v14.0.4`. looking at the package-lock, it looks like hapi `v17.8.5` depends on that.

Screen Shot 2021-08-30 at 11 52 09 AM

**Review by Lianathanoj — APPROVED** [review]
one more comment but LGTM once that's resolved!

File: package.json (hunk 1 above)
Comment (sowmitranalla) [reply package.json]: done!

**Comment by tlhunter** [comment]
I'll modify the repo to require squash and merge

**Pull request merged**